ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -298,6 +298,17 @@ protected SchemaKTable buildTable(final PlanNode node) {
         throw new RuntimeException("Expected to find a Table, found a stream instead.");
       }
 
+      if (schemaKStream.getKeyField() != null
+          && !SchemaUtil.matchFieldName(schemaKStream.getKeyField(), keyFieldName)) {
+        throw new KsqlException(
+            String.format(
+                "Source table key column (%s) is not the column used in the join criteria (%s).",
+                schemaKStream.getKeyField().name(),
+                keyFieldName
```

Comment:
Can we also include the table alias in the message. It would make it less ambiguous if we have columns with the same name in both sides of the join.